subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -318,6 +318,8 @@ private void configureCompileDefaults(final Project project, final JavaPluginCon
                 JavaCompile javaCompile = (JavaCompile) compile;
                 if (javaCompile.getOptions().getRelease().isPresent()) {
                     return JavaVersion.toVersion(javaCompile.getOptions().getRelease().get()).toString();
+                } else if (javaCompile.getJavaCompiler().isPresent()) {
```

Comment:
I knew I fixed this before

https://github.com/gradle/gradle/pull/14425/files#diff-3c55da711188fd617ee234253a70b6da

Reply:
And btw good to know for you: there was an existing bug using the release flag. With the release flag set, you may end up using the wrong java version (without toolchains involved): https://github.com/gradle/gradle/pull/14425/files#diff-e970d36a1f73d09c05bdd69a7a585f92R323